### DISCUSSION TITLE
depth is always zero from three 134->135

### DISCUSSION THREAD

**Discussion by alexpineda** [pmndrs/postprocessing#341] [Q&A] [answered]
I'm currently in the process of migrating to 133 to 136 in three. From 134->135 my custom effect no longer functions because the depth value coming from main differs

```js
void mainImage(const in vec4 inputColor, const in vec2 uv, const in float depth, out vec4 outputColor) {
```

I'm wondering if anyone knows what changed in three from this version. I've poked around and see a lot of changes in renderer nothing I really understand, and their migration guides don't mention anything in particular.

My implementation: https://github.com/imbateam-gg/titan-reactor/tree/dev/packages/titan-reactor/src/renderer/fogofwar

I noticed in 6.23.4 that there is a comment:
`Changed format of noise textures to RGBFormat to avoid issues with LuminanceFormat.`

My [fog texture](https://github.com/imbateam-gg/titan-reactor/blob/dev/packages/titan-reactor/src/renderer/fogofwar/fog-of-war.ts#L22) happens to be `LuminanceFormat`. Is there some other single byte format I should be using?

**Comment by vanruesc** [comment, marked as answer]
> I'm wondering if anyone knows what changed in three from this version.

I haven't investigated this yet, but depth texture support might be broken when multisample anti-aliasing is enabled (via `multisampling`). The reason for this could be that texture management was changed to use `gl.texStorage2D()` in `three` r135 due to issues with `SRGB8_ALPHA8`.

> My fog texture happens to be LuminanceFormat. Is there some other single byte format I should be using?

You can use `RedFormat` if you're using WebGL 2. Otherwise, just stick to `RGBAFormat` because `three` recently removed support for `RGBFormat`.

**Comment by alexpineda** [comment]
Thanks for your response. `multisampling` at 0 did not rectify the issue unfortunately.

**Comment by alexpineda** [comment]
The latest version `6.24.0` has resolved this thank you such much. 🎆 👍

**Comment by alexpineda** [comment]
![ezgif-5-c2c910945a](https://user-images.githubusercontent.com/586716/153120765-4fa4faf4-0e46-42b9-ba08-10ab5ace2f20.gif)


just wanted to showcase some of what I'm doing with postprocessing, using depth of field and scanline 👍

**Reply by vanruesc** [reply]
Nice, thanks for sharing!
